enhancements/proposals/1277-expose-scyllacluster/README.md

```diff
@@ -0,0 +1,398 @@
+    HeadlessServiceType NodeServiceType = "Headless"
+    
+    // ClusterIPServiceType means nodes will be exposed via ClusterIP Service.
+    ClusterIPServiceType NodeServiceType = "ClusterIP"
+    
+    // LoadBalancerServiceType means nodes will be exposed via LoadBalancer Service.
+    LoadBalancerServiceType NodeServiceType = "LoadBalancer"
+)
+
+type NodeServiceTemplate struct {
+    // annotations is a custom key value map merged with every node Service annotations.
+    // +optional
+    Annotations map[string]string `json:"annotations,omitempty"`
+    
+    // type is the Kubernetes Service type.
+    // +kubebuilder:validation:Required
+    Type NodeServiceType `json:"type"`
+    
+    // externalTrafficPolicy controls value of corev1/Service.spec.externalTrafficPolicy of each node Service.
+    // Check Kubernetes corev1.Service documentation about semantic of this field.
+    // +optional
+    ExternalTrafficPolicy *corev1.ServiceExternalTrafficPolicy `json:"externalTrafficPolicy,omitempty"`
+    
+    // allocateLoadBalancerNodePorts controls value of corev1/Service.spec.allocateLoadBalancerNodePorts of each node Service.
+    // Check Kubernetes corev1.Service documentation about semantic of this field.
+    // +optional
+    AllocateLoadBalancerNodePorts *bool `json:"allocateLoadBalancerNodePorts,omitempty"`
+    
+    // loadBalancerClass controls value of corev1/Service.spec.loadBalancerClass of each node Service.
+    // Check Kubernetes corev1.Service documentation about semantic of this field.
+    // +optional
+    LoadBalancerClass *string `json:"loadBalancerClass,omitempty"`
+    
+    // internalTrafficPolicy controls value of corev1/Service.spec.internalTrafficPolicy of each node Service.
+    // Check Kubernetes corev1.Service documentation about semantic of this field.
+    // +optional
+    InternalTrafficPolicy *corev1.ServiceInternalTrafficPolicy `json:"internalTrafficPolicy,omitempty"`
+}
+
+// ExposeOptions hold options related to exposing ScyllaCluster backends.
+type ExposeOptions struct {
+    // nodeService controls properties of Service dedicated for each ScyllaCluster node.
+    // +kubebuilder:default:={type:"ClusterIP"}
+    NodeService *NodeServiceTemplate `json:"nodeService,omitempty"`
+    
+    // BroadcastOptions defines how ScyllaDB node publishes its IP address to other nodes and clients
+    BroadcastOptions *NodeBroadcastOptions `json:"broadcastOptions,omitempty"`
+    
+    // ListenOptions defined how ScyllaDB node listens on interfaces.
+    ListenOptions *NodeListenOptions `json:"listenOptions,omitempty"`
+}
+```
+
+Scylla Operator will create dedicated node Services based on user provided `NodeServiceTemplate`.  
+ScyllaDB configuration parameters will be mapped accordingly:
+* `broadcast_address` - from `BroadcastOptions.Nodes`.
+* `broadcast_rpc_address` - from `BroadcastOptions.Clients`.
+* `rpc_address` - from `ListenOptions.Clients`.
+* `listen_address` - from `ListenOptions.Nodes`.
```

Comment:
At the moment we're setting `listen_address` to `0.0.0.0`. Could you elaborate on why is this change required and whether it has any implications? The use cases above don't mention the listen address at all.

Reply:
Added use case and more mentions about it.